FaunaDB.Client/Client/DefaultClientIO.cs

```diff
@@ -24,13 +24,15 @@ internal class DefaultClientIO : IClientIO
         private readonly HttpClient client;
         private readonly AuthenticationHeaderValue authHeader;
 
+        private readonly IReadOnlyDictionary<string, string> customHeaders;
+
         private LastSeen lastSeen;
         private Version httpVersion;
 
         public const string StreamingPath = "stream";
         public const HttpMethodKind StreamingHttpMethod = HttpMethodKind.Post;
 
-        internal DefaultClientIO(HttpClient client, AuthenticationHeaderValue authHeader, LastSeen lastSeen, Uri endpoint, TimeSpan? timeout, Version httpVersion)
+        internal DefaultClientIO(HttpClient client, AuthenticationHeaderValue authHeader, LastSeen lastSeen, Uri endpoint, TimeSpan? timeout, Version httpVersion, IReadOnlyDictionary<string, string> customHeaders)
         {
```

Comment:
I don't think is necessary for the current PR, but WDYT about adding some builders to replace those methods/constructors w/ big arguments list? We have 7 here

Reply:
Yes, you're right. We actually use a builder in Java driver, it makes sense to it here as well.